src/main/resources/webpublic/index.js

```diff
@@ -23,27 +23,51 @@ function showSummary(container, summary) {
 /* Create a box with answer sources. */
 function showSources(container, sources) {
 	container.empty();
-	sources.forEach(function(s) {
+        $.each(sources, function(sid, source) {
 		var state_stags = ['<i>', '<b>', ''];
 		var state_etags = ['</i>', '</b>', ''];
-		container.append('<p class="source">'
-				+ '<img src="/wikipedia-w-logo.png" alt="W" class="wlogo" />'
-				+ ' <a href="http://en.wikipedia.org/?curid='+s.pageId+'" target="_blank">'
-				+ state_stags[s.state] + s.title + state_etags[s.state]
-				+ '</a> (' + s.origin + ')</p>'); // TODO also include the first sentence?
+        if(!(typeof (source.pageId) ==="undefined")) { //this forces to only show en wiki
+            container.append('<p class="source">'
+                + '<img src="/wikipedia-w-logo.png" alt="W" class="wlogo" />'
+                + ' <a href="http://en.wikipedia.org/?curid=' + source.pageId + '" target="_blank">'
+                + state_stags[source.state] + source.title + state_etags[source.state]
+                + '</a> (' + source.origin + ')</p>'); // TODO also include the first sentence?
+        }
 	});
+
 }
 
 /* Create a table with answers. */
-function showAnswers(container, answers, passages) {
+function showAnswers(container, answers, snippets,sources) {
 	container.empty();
 	var i = 1;
 	answers.forEach(function(a) {
 		// FIXME: also deal with < > &
 		text = a.text.replace(/"/g, "&#34;");
-        var str="found using "+ a.source+" \n";
-        for(var index = 0; index< a.passageIDs.length; index++) {
-         str += " "+index+". "+passages[a.passageIDs[index]].replace(/"/g, "&#34;")+"\n";
+        var str="";
+
+
+        for(var index = 0; index< a.snippetIDs.length; index++) {
+            //origin is (fulltext)/(title-in-clue)/(documented search)
+            str +="("+sources[snippets[a.snippetIDs[index]].sourceID].origin+") \n";
+            str +=sources[snippets[a.snippetIDs[index]].sourceID].title + " \n";
+
+
+            //add either wikipedia document ID or source URL
+            if(!(typeof(sources[snippets[a.snippetIDs[index]].sourceID].pageId )==="undefined")) {
+                str += "http://en.wikipedia.org/?curid=" + sources[snippets[a.snippetIDs[index]].sourceID].pageId + "\n";
+            }
+            else if(!(typeof(sources[snippets[a.snippetIDs[index]].sourceID].URL )==="undefined")){
+                str+= sources[snippets[a.snippetIDs[index]].sourceID].URL+ "\n";
+            }
+
+            //add either passage text or property label
+            if (!(typeof (snippets[a.snippetIDs[index]].passageText) ==="undefined")) {
+                str += snippets[a.snippetIDs[index]].passageText.replace(/"/g, "&#34;") + "\n";
+            }
+            else if (!(typeof (snippets[a.snippetIDs[index]].propertyLabel) ==="undefined")) {
+                str += snippets[a.snippetIDs[index]].propertyLabel + "\n";
+            }
```

Comment:
Style: Strange indentation, spaces _around_ +=, between if and (, around + ... (not important tho')